src/main/java/techreborn/blockentity/machine/tier0/block/BlockPlacerBlockEntity.java

```diff
@@ -0,0 +1,138 @@
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+import reborncore.client.screen.builder.BlockEntityScreenHandlerBuilder;
+import reborncore.client.screen.builder.ScreenHandlerBuilder;
+import reborncore.common.blockentity.MachineBaseBlockEntity;
+import reborncore.common.recipes.RecipeCrafter;
+import reborncore.common.screen.BuiltScreenHandler;
+import reborncore.common.screen.BuiltScreenHandlerProvider;
+import reborncore.common.util.RebornInventory;
+import techreborn.blockentity.machine.GenericMachineBlockEntity;
+import techreborn.blockentity.machine.tier0.block.blockplacer.BlockPlacerProcessor;
+import techreborn.config.TechRebornConfig;
+import techreborn.init.TRBlockEntities;
+import techreborn.init.TRContent;
+
+/**
+ * <b>A machine that can break place blocks in front of it</b>
+ * <br>
+ * May not place blocks with hardness < 0.
+ * Shares most functionality with other GenericMachineBlockEntities, however does not make use of the {@code RecipeCrafter}
+ *
+ * @author SimonFlapse
+ */
+public class BlockPlacerBlockEntity extends GenericMachineBlockEntity implements BuiltScreenHandlerProvider, BlockProcessable {
+
+	public static final int BASE_BREAK_TIME = TechRebornConfig.blockPlacerBaseBreakTime;
+	public static final int COST_TO_BREAK = TechRebornConfig.blockPlacerEnergyPerTick;
+	public static final int ENERGY_SLOT = 0;
+	public static final int INPUT_SLOT = 1;
+	public static final int FAKE_OUTPUT_SLOT = 2;
+
+	private final BlockPlacerProcessor processor = new BlockPlacerProcessor(this, INPUT_SLOT, FAKE_OUTPUT_SLOT, BASE_BREAK_TIME, COST_TO_BREAK);
+
+	public BlockPlacerBlockEntity(BlockPos pos, BlockState state) {
+		super(TRBlockEntities.BLOCK_PLACER, pos, state, "Block Placer", TechRebornConfig.blockPlacerMaxInput, TechRebornConfig.blockPlacerMaxEnergy, TRContent.Machine.BLOCK_PLACER.block, ENERGY_SLOT);
+		this.inventory = new RebornInventory<>(3, "BlockPlacerBlockEntity", 64, this);
+	}
+
+	// BuiltScreenHandlerProvider
+	@Override
+	public BuiltScreenHandler createScreenHandler(int syncID, PlayerEntity player) {
+		BlockEntityScreenHandlerBuilder builder = new ScreenHandlerBuilder("blockplacer")
+			.player(player.getInventory())
+			.inventory()
+			.hotbar()
+			.addInventory()
+			.blockEntity(this)
+			.energySlot(ENERGY_SLOT, 8 , 72).syncEnergyValue()
+			.slot(INPUT_SLOT, 80, 60);
+
+			processor.syncNbt(builder);
+
+			return builder.addInventory().create(this, syncID);
+	}
+
+	@Override
+	public void tick(World world, BlockPos pos, BlockState state, MachineBaseBlockEntity blockEntity) {
+		super.tick(world, pos, state, blockEntity);
+		if (world == null || world.isClient) {
+			return;
+		}
+
+		processor.onTick(world, getFrontBlockPosition(blockEntity.getFacing(), pos));
+	}
+
+	private BlockPos getFrontBlockPosition(Direction direction, BlockPos pos) {
```

Comment:
Just being curious but what is this for? pos.offset(blockEntity.getFacing() == null? Direction.NORTH : blockEntity.getFacing()) should do the work if its checking NPE

Reply:
`Pos.offset` may very well work :) I'm still getting used to he APIs.

Thanks for suggesting